ssv/spectest/tests/committee/past_msg_duty_finished.go

```diff
@@ -1,11 +1,119 @@
+	ksMap := testingutils.KeySetMapForValidators(numValidators)
+	ks := testingutils.Testing4SharesSet()
+
+	decidedValue := testingutils.TestBeaconVoteByts
+	msgID := testingutils.CommitteeMsgID(ks)
+	pastHeight := qbft.Height(10)
+
+	bumpHeight := func(c *ssv.Committee, previousDuty types.Duty) *ssv.Committee {
+
+		err := c.StartDuty(previousDuty.(*types.CommitteeDuty))
+		if err != nil {
+			panic(err)
+		}
+
+		happyFlowMessages := []*types.SignedSSVMessage{
+			testingutils.TestingProposalMessageWithIdentifierAndFullData(
+				ks.OperatorKeys[1], types.OperatorID(1), msgID, decidedValue,
+				pastHeight),
+			testingutils.TestingPrepareMessageWithParams(ks.OperatorKeys[1], 1, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+			testingutils.TestingPrepareMessageWithParams(ks.OperatorKeys[2], 2, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+			testingutils.TestingPrepareMessageWithParams(ks.OperatorKeys[3], 3, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+
+			testingutils.TestingCommitMessageWithParams(ks.OperatorKeys[1], 1, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+			testingutils.TestingCommitMessageWithParams(ks.OperatorKeys[2], 2, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+			testingutils.TestingCommitMessageWithParams(ks.OperatorKeys[3], 3, qbft.FirstRound, pastHeight, msgID, sha256.Sum256(decidedValue)),
+
+			testingutils.SignPartialSigSSVMessage(ks, testingutils.SSVMsgCommittee(ks, nil, testingutils.PostConsensusAttestationMsgForKeySet(ksMap, 1, pastHeight))),
+			testingutils.SignPartialSigSSVMessage(ks, testingutils.SSVMsgCommittee(ks, nil, testingutils.PostConsensusAttestationMsgForKeySet(ksMap, 2, pastHeight))),
+			testingutils.SignPartialSigSSVMessage(ks, testingutils.SSVMsgCommittee(ks, nil, testingutils.PostConsensusAttestationMsgForKeySet(ksMap, 3, pastHeight))),
+		}
+		for _, msg := range happyFlowMessages {
+			err := c.ProcessMessage(msg)
+			if err != nil {
+				panic(err)
+			}
+		}
+
+		// Erase broadcasted messages and roots due to test setup
+		c.Runners[previousDuty.DutySlot()].GetNetwork().(*testingutils.TestingNetwork).BroadcastedMsgs = make([]*types.SignedSSVMessage, 0)
+		c.Runners[previousDuty.DutySlot()].GetBeaconNode().(*testingutils.TestingBeaconNode).BroadcastedRoots = make([]phase0.Root, 0)
+
+		return c
+	}
+
+	pastProposalMsgF := func() *types.SignedSSVMessage {
+		fullData := decidedValue
+		root, _ := qbft.HashDataRoot(fullData)
+		msg := &qbft.Message{
+			MsgType:    qbft.ProposalMsgType,
+			Height:     pastHeight,
+			Round:      qbft.FirstRound,
+			Identifier: msgID,
+			Root:       root,
+		}
+		signed := testingutils.SignQBFTMsg(ks.OperatorKeys[1], 1, msg)
+		signed.FullData = fullData
+
+		return signed
+	}
+
+	expectedError := "failed processing consensus message: could not process msg: invalid signed message: proposal is not valid with current state"
```

Comment:
looking at the code I would think the error should be "wrong msg height"

Reply:
But in this runner the height is correct. Remember that a message is redirected to the runner that is assigned to the message's slot.